### PR TITLE
✨feat(config) add option to customize commit message

### PR DESCRIPTION
**Pull request by AndreAugustoAAQ**
I made the necessary changes so that the commit message can be customized

The code is based on this PR #32 and this discussion #51

Related to this PR
#51 #52 #43 #31 #32

### CODE REVIEW

**Comment by AndreAugustoAAQ** [comment]
Hey @ngryman
I made the necessarie changes
- I didn't change the formatting, I used the prettierrc file from the repository
- Removed changes to lib/types.json
- Sorry for the forced push, I had problems with the local repo and redid the changes in new commits and replaced my remote

File: index.js [hunk 1]
```diff
@@ -180,14 +184,14 @@ function format(answers) {
...
     cz.prompt.registerPrompt('maxlength-input', require('inquirer-maxlength-input-prompt'))
 
     getConfig()
       .then(createQuestions)
       .then(cz.prompt)
-      .then(format)
+      .then((answers) => format(answers, configs))
       .then(commit)
-  }
+  },
```
Comment (ngryman) [inline index.js:196]: I think we can move away from piping functions here.

Assuming we add `async` to the parent function, we can do:
```js
const config = await getConfig()
const questions = createQuestions(config)
const answers = cz.prompt(questions)
const commitMessage = formatCommitMessage(answers, config)

commit(commitMessage)
```

File: index.js [hunk 2]
```diff
@@ -155,23 +147,35 @@ function createQuestions(config) {
  * Format the git commit message from given answers.
  *
  * @param {Object} answers Answers provide by `inquier.js`
+ * @param {Object} config Result of the `getConfig` returned promise
  * @return {String} Formated git commit message
  */
-function format(answers) {
+
+function format(answers, config) {
```
Comment (ngryman) [inline index.js:154]: For clarity, I think we should rename this function:
```suggestion
function formatCommitMessage(answers, config) {
```

File: index.js [hunk 3]
```diff
@@ -155,23 +147,35 @@ function createQuestions(config) {
...
+  const emoji = answers.type
+  const type = types.find((type) => type.code === emoji.emoji).name
+  const scope = answers.scope ? '(' + answers.scope.trim() + ')' : ''
+  const subject = answers.subject.trim()
+
+  const commitMessage = format
+    .replace(/\{emoji\}/g, emoji.emoji)
+    .replace(/\{type\}/g, type)
+    .replace(/\{scope\}/g, scope)
+    .replace(/\{subject\}/g, subject)
```
Comment (ngryman) [inline index.js:168]: Minor, for simplicity I would use `replaceAll` with strings instead:

```suggestion
  const commitMessage = format
    .replaceAll('{emoji}', emoji.emoji)
    .replaceAll('{type}', type)
    .replaceAll('{scope}', scope)
    .replaceAll('{subject}', subject)
```

File: index.js [hunk 4]
```diff
@@ -29,15 +29,20 @@ async function getConfig() {
     symbol: false,
     skipQuestions: [''],
     subjectMaxLength: 75,
-    conventional: false
+    format: '{emoji} {scope} {subject}',
```
Comment (ngryman) [inline index.js:32]: Removing `conventional` is a breaking change I think we could avoid. I would put it back and handle it in `format` (cf. comment below).

File: index.js [hunk 5]
```diff
@@ -155,23 +147,35 @@ function createQuestions(config) {
  * Format the git commit message from given answers.
  *
  * @param {Object} answers Answers provide by `inquier.js`
+ * @param {Object} config Result of the `getConfig` returned promise
  * @return {String} Formated git commit message
  */
-function format(answers) {
+
+function format(answers, config) {
+  const format = config.format
```
Comment (ngryman) [inline index.js:155]: cf. comment about `conventional`. When set to `true` it should take precedence over the `config.format`:
```suggestion
  const format = config.conventional
    ? `{type}{scope}: {emoji} {subject}`
    : config.format
```

**Review by ngryman — COMMENTED** [review]
@AndreAugustoAAQ Thanks for the changes. I've got a couple of suggestions left and we're good to merge.

Could you also add `"trailingComma": "none"` to the `.prettierrc` file? It looks like the default is now `es5`.

Thank you!

**Review by ngryman — COMMENTED** [review]
@AndreAugustoAAQ Thanks for the changes. I've got a couple of suggestions left and we're good to merge.

Could you also add `"trailingComma": "none"` to the `.prettierrc` file? It looks like the default is now `es5`.

Thank you!

**Comment by AndreAugustoAAQ** [comment]
Hey @ngryman,
I'm trying to make the proposed changes, but I'm not managing to make it work as expected
is giving error "Cannot read property 'replaceAll' of undefined"
I believe that "types" is not being passed, and I have no idea how to pass it 😅

**Comment by ngryman** [comment]
Hey @AndreAugustoAAQ, no worries! Can you push your changes as is? I can take a look.

**Comment by AndreAugustoAAQ** [comment]
@ngryman I pushed on the dev-fixes branch https://github.com/AndreAugustoAAQ/cz-emoji/tree/dev-fixes

**Comment by luizcorreia** [comment]
Hi, @ngryman what needs to this be done? 
Can I help with somethink? 
I really want this PR to make this compatible with Conventional Commits

**Comment by rossignolli** [comment]
Anything new about this? i really want this PR too.

**Comment by ngryman** [comment]
@AndreAugustoAAQ Could you switch the base branch from `master` to `dev-fixes` so we can have your latest changes? Thanks!

**Comment by ngryman** [comment]
Closing in favor of #66.

**Pull request closed without merging**